### PR TITLE
feat(MenuWithCombobox): add MenuWithCombobox component

### PR DESCRIPTION
**Pull request by balzdur**
First PR required for the refactoring of operand edit : introduce a new component Menu + Combobox (in order to drop the quick & dirty previous composition which was hacky).

PS: look at the storybook to see some example (behaviours: especially keyboard + focus managment + dismissing management)

### CODE REVIEW

File: packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  const parent = React.useContext(ParentContext);
+  const withCombobox = searchValue !== undefined || onSearch !== undefined;
+
+  const element = (
+    <Ariakit.MenuProvider
+      showTimeout={0}
+      placement={parent ? 'right-start' : 'bottom-start'}
+    >
+      <WithComboboxContext.Provider value={withCombobox}>
+        <HideAllContextProvider>{children}</HideAllContextProvider>
```
Comment (Pascal-Delange) [inline packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx:57]: what does this do exactly? Hide context providers from screen readers ?

File: packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
+import * as Ariakit from '@ariakit/react';
```
Comment (Pascal-Delange) [inline packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx:1]: is this an alternative to radix ?

File: packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx (hunk 1 above)
Comment (balzdur) [reply packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx:57]: This is a context I created in order to inject a hideAll method.
This allows to close all menu and submenu at once

**Review by Pascal-Delange — APPROVED** [review]

File: packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx (hunk 2 above)
Comment (balzdur) [reply packages/ui-design-system/src/MenuWithCombobox/MenuWithCombobox.tsx:1]: Yes, TBH I prefer a little bit the simplicity of Radix (composition first with hidden complexity) but Radix lag behind in term of new features and customization is sometimes really hard.

With that new pkg it's simpler (and there is a lot of concrete examples to allow complex components creation). I already used it for the SelectWithCombobox.

As an example, Radix don't have yet any combobox, but the open issue on the topic started 3/4 years ago. This is definitely a pain point for me. Composition of existing components is hard in radix but easier on Ariakit

**Pull request merged**